src/pcapi/repository/user_queries.py

```diff
@@ -29,6 +29,11 @@ def find_user_by_email(email: str) -> User:
     return User.query.filter(func.lower(User.email) == format_email(email)).first()
 
 
+def find_beneficiary_users_by_email_provider(email_provider: str) -> List[User]:
+    formatted_email_provider = "%{}%".format(email_provider)
```

Comment:
Si le but est de filtrer sur le domaine de l'adresse mail, autant être spécifique et chercher `@{email_provider}`. Je suggère même un % entre "@" et `email_provider` pour aussi détecter les sous-domaines, au cas où :

```suggestion
    formatted_email_provider = f"%@%{email_provider}"
```